router/src/infer/mod.rs

```diff
@@ -33,6 +33,11 @@ pub trait Backend {
     ) -> Result<UnboundedReceiverStream<Result<InferStreamResponse, InferError>>, InferError>;
 
     async fn health(&self, current_health: bool) -> bool;
+
+    /// The state of the health on startup
+    /// Typically false, or true if the backend includes
+    /// a warmup phase.
+    fn start_health(&self) -> bool;
```

Comment:
Would be great to default to false.
Otherwise LGTM.

Reply:
Doesn't that make it easy to miss to implement for implementors ?

It's like matching enums, I tend to have exhaustive matches, it avoid silently doing something wrong.